### ISSUE TITLE
Fetching User objects for each company's studentContact in list view

### ISSUE THREAD

**Issue by finnss** [webkom/lego-webapp#568]
**tl;dr:** I don't know how I should fetch User objects for each company to render its studentContact properly, particularily in List view.

I'm a bit baffled by this issue, and I'm almost certain there's a really easy solution that I'm not seeing.

It seems that fetching companies, both list and detail, results in a payload containing not only the companies but also the full User objects for each company's student contact (which is just the id by default). I'm not sure where this is configured to happen, but it seems useful. I need access to that User in both Bdb list view and detail view. However, I don't know the correct way to use it.

My first idea was to get the users into the global state and insert them each company in `selectCompanies`. However, I don't think you're allowed to set global state for anything else than your own Entity. 
```
function mutateCompanies(state, action) {
  switch (action.type) {
    case Company.DELETE.SUCCESS: {
       ...
```

This is the function I use to reduce companies. Logging `state` yields only companies and neighbouring fields like `actionGrant`, which makes me think I'm unable to change anything "above" that, and thus unable to put my Users (available through `action.payload.entities.users`) into the global state.

Since that didn't work, I figured I had to put the users into the companies directly in these reducers. I did something like (for the detail fetch):
```
    case Company.FETCH.SUCCESS: {
      return {
        ...state,
        byId: {
          ...state.byId,
          studentContact: action.payload.entities.users[action.payload.result]
        }
      }
    }
``` 

Not this exact code, but the idea worked. I was happy. However, I realized that if I want to rely on these reducers to get me the User objects, I would have to implement similar logic in `FETCH_ALL`, `EDIT` and `ADD`. Even with only the example, that would mean overriding all the nice default reducers and creating a huge hassle. This could work though, and if people think that's the way we should do then fine. I wasn't confident that was the case, so I didn't want to go through all that reducer pain until I was sure.

So instead, I added the following to my `fetch` function:
```
.then(response => {
      response.payload &&
        dispatch(
          fetchUser(
            response.payload.entities.companies[companyId].studentContact
          )
        );
    })
```
This puts the User into state the way I originally wanted, and selecting it from there works fine. However, I couldn't bring myself to fire off a similar fetch for every company in the list view, and I didn't know if there was any kind of 
```
callAPI({
        types: User.FETCH,
        endpoint: `/user/?ids=[1, 2, 4]/`,
```
So for now, I just display the username in the list view. Would like to get access to the full objects.
